pkg/ls/compile.go

```diff
@@ -245,6 +248,60 @@ func (compiler *Compiler) compile(context *Context, ctx *compilerContext, ref st
 	return compiled, nil
 }
 
+func (compiler *Compiler) compileIncludeAttribute(context *Context, ctx *compilerContext) error {
+	nodeMap := make(map[*lpg.Node]*lpg.Node)
+	copySubtree := func(targetNode, includeNode *lpg.Node, tgtGraph *lpg.Graph) error {
+		nodeMap[includeNode] = targetNode
+		for edges := includeNode.GetEdges(lpg.OutgoingEdge); edges.Next(); {
+			edge := edges.Edge()
+			srcTypes := lpg.NewStringSet(FilterAttributeTypes(targetNode.GetLabels().Slice())...)
+			includeTypes := FilterAttributeTypes(includeNode.GetLabels().Slice())
+			for _, typ := range includeTypes {
+				if !srcTypes.Has(typ) {
+					return ErrNotFound(fmt.Sprintf("attribute type for source: %v do not match with include: %v", targetNode, includeNode))
+				}
+				srcTypes.Add(typ)
+			}
+			targetNode.SetLabels(srcTypes)
+			includeNamespace := GetAttributeID(edge.GetTo())
```

Comment:
You have to change the id of the includeNode such that, the new id becomes:

namespace/lastComponent

where namespace comes from the "namespace" tag of the targetNode

Before copySubtree, get the namespace attribute from the targetNode, and use that all over. If empty, no namespace rewriting.

If the attrId does not have "/", do not rewrite the namespace